fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -95,50 +95,58 @@ public void ensureValid() {
         }
 
 
-        private static byte[] fileLockValue(long timeStampMillis) {
-            return Tuple.from(timeStampMillis).pack();
+        private byte[] fileLockValue() {
+            return Tuple.from(selfStampUuid, timeStampMillis).pack();
```

Comment:
If you flipped the two around, so that `timeStampMillis` is first, I think this would be backwards compatible. Probably not super important given the current state of the lucene index, but something to keep in mind for future changes.